### DISCUSSION TITLE
Steps to run dnsproxy and wgproxy

### DISCUSSION THREAD

**Discussion by dd76** [celzero/firestack#25] [Q&A] [answered]
Hi Guys, 

How do I run just the dnsproxy and wgproxy for android plz help. 

Regards,
dd

**Comment by ignoramous** [comment, marked as answer]
Sorry, can you be a bit more elaborate?

If you're looking to setup WireGuard with other VPN providers, then see below:

---

The UI in Rethink is borrowed *as-is* from [the official WireGuard Android app](https://github.com/wireguard/wireguard-android). Following *any* tutorial for the *official* WireGuard app should work for Rethink, too.

Ex: [mullvad](https://archive.is/pe1NK), [ivpn](https://archive.is/4Sce3), [windscribe](https://archive.is/nboff).

After you've a QR code, a configuration file, or settings for WireGuard ready-to-go, Rethink specific steps are:

1. Open Rethink, goto Configure -> Proxy -> Setup WireGuard
   - Tap on the "+" button
   - Choose Import / QR code / Create as approp
   - Follow VPN-specific *WireGuard for Android* setup guides.
2. If the previous steps are successful, you should see a *new* profile prefixed `wg` added to the list on Rethink.
3. Tap on that new `wg` entry.
   - In the new screen that opens up, tap on Add / Remove applications.
   - Select apps that should be part of this particular WireGuard profile.
   - Go back.
4. Enable / disable `wg` profile as you will.

Also see: https://www.reddit.com/r/rethinkdns/comments/163e83l/proton_vpn/

---

From:  https://www.reddit.com/r/rethinkdns/comments/161xkt7/how_to_make_wireguard_work_with_rethinkdns/
